### PR TITLE
Surf props

### PR DESCRIPTION
**Pull request by pshriwise**
## Description
This PR includes additional options for surface boundary conditions to the `dagmcMetaData` class. It also ignores capitalization when comparing strings for boundary condition values as well as the graveyard and vacuum identifiers.

A test case for the lowercase values as well as a new boundary condition has been added in the `TestBoundaryAssigns` test.

## Motivation and Context
This allows for support for other commonly used boundary conditions in Monte Carlo codes.

### CODE REVIEW

File: src/dagmc/dagmcmetadata.cpp [hunk 1]
```diff
@@ -337,10 +345,17 @@ void dagmcMetaData::parse_boundary_data() {
     // 2d entities have been tagged with the boundary condition property
     // ie. both surfaces and its members triangles,
 
-    if (boundary_assignment[0].find("Reflecting") != std::string::npos)
+
+    std::string bc_string = to_lower(boundary_assignment[0]);
+
+    if (bc_string.find("reflecting") != std::string::npos)
```
Comment (gonuke) [inline src/dagmc/dagmcmetadata.cpp]: Does it make sense to unmagic these strings like we've done with `graveyard_str` and `vacuum_str`?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @pshriwise - nice additions.

There may be robustness questions with a DAGMC periodic boundary, but look forward to seeing an implementation.

File: src/dagmc/dagmcmetadata.cpp (hunk 1 above)
Comment (pshriwise) [reply src/dagmc/dagmcmetadata.cpp]: Yeah I'd say so. Can do!

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
LGTM - Thanks @pshriwise
